### ISSUE TITLE
FOUC (Flash Of Unstyled Content)

### ISSUE THREAD

**Issue by 1rens1** [mantinedev/mantine#1148]
### What package has an issue

@mantine/core

### Describe the bug

Hello,
I've been using mantine for quite some time and I just first realized that there was a flash of unstyled content, I don't know how I missed it but it's there alright.
The flash is not too long, but still.
I've been looking around how mantine inject/add stylesheets to the dom. Turns out it adds style elements to the dom per elements
![image](https://user-images.githubusercontent.com/78131273/161214468-ec508259-21ed-42f6-ac79-66cbb3ee946c.png)

![image](https://cdn.has-cool.pics/P2ADqZReU6.gif)

### In which browser did the problem occur

Microsoft Edge

### Do you know how to fix the issue

No

### Are you willing to participate in fixing this issue and create a pull request with the fix

No

### Possible fix

Render the style elements on the server and not the client.

**Comment by rtivital** [comment]
You need to setup server side rendering on your side. You can find full tutorials on Getting started page – https://mantine.dev/getting-started/

**If you are using Next.js**

Use template (https://github.com/mantinedev/mantine-next-template) or:
- Follow the Next.js getting started guide – https://mantine.dev/theming/next/
- Delete .next folder to reset cache

Note that Mantine does not support experimental versions of react and next. If something does not work for you – use template (https://github.com/mantinedev/mantine-next-template).

**Comment by 1rens1** [comment]
@rtivital
Thanks, I didn't set up any fancy stuff on my app, I just `npx create-next-app` install some mantine stuff and imported some other stuff lol

**Comment by ilomon10** [comment]
@1rens1 how do you solve this FOUC problem?

**Comment by 1rens1** [comment]
@ilomon10 Look [above](https://github.com/mantinedev/mantine/issues/1148#issuecomment-1085528609)

**Comment by ilomon10** [comment]
@1rens1 Thank you, i just resolve my problem by removing custom emotionOptions at MantineProvider, i dont know why but maybe @rtivital can explain it. Please.

**Comment by major-snow** [comment]
again in nextjs v12.1.6

**Comment by rtivital** [comment]
Yeah, there are some issues with 12.1.6 patch

**Comment by kwiat1990** [comment]
I have tried everything posted here and in other threads but none of solution have really worked for me. As the last step I have forked [this Next repo](https://github.com/mantinedev/mantine-next-template) from @rtivital and clean `package.json` up - some packages were removed, some were added and then I upgrade everything to the latest versions (expect for Next, which I left by version `12.1.4`). After that I started to migrate all my code step by step, while checking if FOUC comes. And I'm happy to announce that it didn't.
